### DISCUSSION TITLE
relax.Var和tvm.te.tensor.Tensor的区别是什么？

### DISCUSSION THREAD

**Discussion by hbsun2113** [mlc-ai/mlc-zh#90] [Q&A] [answered]
请问这里`C`的类型是`relax.Var`而不是`tvm.te.tensor.Tensor`，是出于什么考虑？
![image](https://user-images.githubusercontent.com/27994442/182028242-703cbbbc-b44b-4b3f-b568-c877385530e4.png)

**Comment by yzh119** [comment, marked as answer]
`tvm.te.tensor.Tensor`是运行时对象，`relax.Var`是relax IR里的一个节点，并不直接与一个运行时对象绑定。
在MLC中我们处理的都是符号类型对象而非实体的矩阵类，如果C的类型是`tvm.te.tensor.Tensor`，那意味着我们需要执行C=A*B这一计算才能得到C这个对象，这是pytorch这类深度学习框架中eager模式的思维方式。
MLC执行的流程是：把整个IRModule编译成一个函数，编译出来的函数接收运行时Tensor类型的A和B作为参数，输出计算结果D。
在编译之前，操作的所有对象都是IR里的节点，编译之后我们才会与运行时对象进行交互。
